File: climbing-stairs/climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        
        if n == 1:
            return 1
        fib = [0]*n
        fib[0] = 1
        fib[1] = 2
        
        for i in range(2,n):
            fib[i] = fib[i-1] + fib[i-2]
            
        #print(fib)
        return fib[-1]
        

File: climbing-stairs/climbing-stairs.java
class Solution {
    public int climbStairs(int n) {
     
        int step1 = 1;
        int step2 = 1;
        
        if(n <= 1){ // base case 
            return step1;
        }
        
        int res = 0;
        
        for(int i=1;i<n;i++){ // run until n create fibnacci
            res = step1 + step2;
            step2 = step1;
            step1 = res;
        }
        
        return res;
        
    }
}